integration_tests/tests/integrity_check_fanout.sql

```diff
@@ -0,0 +1,37 @@
+{{ config(
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+{% set tables_names = ['account', 'appointment', 'contact', 'incident', 'opportunity', 'phonecall', 'systemuser'] %}
+
+with source_counts as (
+    {% for table_name in tables_names %}
+        select 
+            count(*) as source_row_count,
+            {{ "'" ~ table_name ~ "'" }} as source_name
+        from {{ source('dynamics_365_crm', table_name) }}
+
+        {{ 'union all ' if not loop.last }}
+    {% endfor %}
+
+), mapped_counts as (
+    {% for table_name in tables_names %}
+        select 
+            count(*) as mapped_row_count,
+            {{ "'" ~ table_name ~ "'" }} as mapped_name
+        from {{ ref(table_name) }}
+
+        {{ 'union all ' if not loop.last }}
+    {% endfor %}
+)
+
+-- Produce rows if any row counts do not match
+select 
+    source_name,
+    source_row_count,
+    mapped_row_count
+from source_counts
+inner join mapped_counts
+    on source_name = mapped_name
+    and source_row_count != mapped_row_count
```

Comment:
More of a preference comment - I find it easier to understand the test criteria when it's separated out as a where clause and not nested within the join condition. 
```suggestion
where source_row_count != mapped_row_count
```

Reply:
Makes sense--udpated.